core/src/storage/alloc/bump_alloc.rs

```diff
@@ -39,7 +39,7 @@ pub struct BumpAlloc {
 impl BumpAlloc {
     /// Creates a new forward allocator for the given raw parts.
     ///
-    /// # Note
+    /// # Safety
     ///
     /// Do not use this directly!
     /// This is meant to be used by pDSL internals only.
```

Comment:
Highly outdated docs. You are right that we should have used the `Safety` section here, however, the safety section is normally used to communicate what to take care and look for if you are about to call it so we should eventually do that here as well.